star_realms/src/game/effects/mod.rs

```diff
@@ -23,14 +23,17 @@ pub mod conditions;
 /// A trait that the client implements for choosing configuration values for
 pub trait ConfigSupplier {
     /// get a config value (u32) for an action based on this Config object
-    fn get_config(&self, game: &GameState, config: &Config) -> u32;
+    fn get_config(&self, game: &GameState, config: &UserConfigMeta) -> u32;
 }
 
 /// I mean this is just what I use for error reporting in general
 pub type ConfigError = String;
 
 /// describes data contained in yaml configuration for action or condition
-pub type PreConfig = HashMap<String, String>;
+pub struct PreConfig {
```

Comment:
Made it a struct so we can validate it using a `PreConfigMeta` to validate typed values during startup

Reply:
or during tests